modules/programs/i3lock.nix

```diff
@@ -0,0 +1,412 @@
+          type = types.nullOr types.int;
+          default = cfg.text.size.default;
+        };
+
+        default = mkOption {
+          description = "i3lock default text size";
+          type = types.nullOr types.int;
+          default = null;
+        };
+      };
+    };
+
+    colors = {
+      insideVerify = mkOption {
+        description = "i3lock inside verify color";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ringVerify = mkOption {
+        description = "i3lock ring verify color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      insideWrong = mkOption {
+        description = "3lock inside wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ringWrong = mkOption {
+        description = "i3lock ring wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      inside = mkOption {
+        description = "3lock inside color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      ring = mkOption {
+        description = "i3lock ring color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      line = mkOption {
+        description = "i3lock line color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      separator = mkOption {
+        description = "i3lock seprarator color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      verify = mkOption {
+        description = "i3lock verify color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      wrong = mkOption {
+        description = "i3lock wrong color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      time = mkOption {
+        description = "i3lock time color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      date = mkOption {
+        description = "i3lock date color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      layout = mkOption {
+        description = "i3lock layout color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      keyhl = mkOption {
+        description = "i3lock key hold color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      bshl = mkOption {
+        description = "i3lock backspace hold color.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+    };
+
+    extraCommands = mkOption {
+      description = "Extra commands to run before running i3lock.";
+      default = "";
+      type = types.lines;
+      example = ''
+        ''${pkgs.scrot}/bin/scrot /tmp/screen_locked.png
+        ''${pkgs.imagemagick}/bin/convert /tmp/screen_locked.png \
+          -scale 10% -scale 1000% /tmp/screen_locked.png
+      '';
```

Comment:
This example seems to render incorrectly in the manual:

```
programs.i3lock.extraCommands
           Extra commands to run before running i3lock.

           Type: strings concatenated with "\n"

           Default: ""

           Example:

               ''
               ''${pkgs.scrot}/bin/scrot /tmp/screen_locked.png
               ''${pkgs.imagemagick}/bin/convert /tmp/screen_locked.png \
                 -scale 10% -scale 1000% /tmp/screen_locked.png
               ''
```

Maybe we should drop the `''` before $?

Reply:
Fixed using `literalExample`.